File: 프로그래머스/JavaScript/Lv. 1/42840. 모의고사/모의고사.js
function solution(answers) {
  let n = [
    [1, 2, 3, 4, 5],
    [2, 1, 2, 3, 2, 4, 2, 5],
    [3, 3, 1, 1, 2, 2, 4, 4, 5, 5],
  ];

  let cnt = [0, 0, 0];
  for (let i = 0; i < answers.length; i++) {
    for (let j = 0; j < 3; j++) {
      if (answers[i] == n[j][i % n[j].length]) {
        cnt[j] += 1;
      }
    }
  }

  const max = Math.max(...cnt);
  let answer = [];
  for (let i = 0; i < cnt.length; i++) {
    if (cnt[i] === max) {
      answer.push(i + 1);
    }
  }
  return answer;
}

File: 프로그래머스/Java/Lv. 1/42840. 모의고사/모의고사.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] answers) {

        int[][] patterns = {
                { 1, 2, 3, 4, 5 },
                { 2, 1, 2, 3, 2, 4, 2, 5 },
                { 3, 3, 1, 1, 2, 2, 4, 4, 5, 5 }
        };

        int[] scores = new int[3];

        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < answers.length; j++) {
                if (answers[j] == patterns[i][j % patterns[i].length]) {
                    scores[i]++;
                }
            }
        }

        int max = scores[0];
        for (int i = 0; i < scores.length; i++) {
            if (scores[i] > max) {
                max = scores[i];
            }
        }

        ArrayList<Integer> list = new ArrayList<>();
        for (int i = 0; i < 3; i++) {
            if (scores[i] == max) {
                list.add(i + 1);
            }
        }

        int[] result = list.stream().mapToInt(Integer::intValue).toArray();

        return result;
    }
}
